tests/baselines/reference/privateNameInInExpressionTransform.js

```diff
@@ -0,0 +1,97 @@
+
+        v << #field in v << v; // Good precedence: (v << (#field in (v << v)))
+
+        v << #field in v == v; // Good precedence: ((v << (#field in v)) == v)
+
+        v == #field in v < v; // Good precedence: (v == ((#field in v) < v))
+
+        #field in v && #field in v; // Good precedence: ((#field in v) && (#field in v))
+    }
+    invalidLHS(v: any) {
+        'prop' in v = 10;
+        #field in v = 10;
+    }
+}
+
+class Bar {
+    #field = 1;
+    check(v: any) {
+        #field in v; // expect Bar's 'field' WeakMap
+    }
+}
+
+function syntaxError(v: Foo) {
+    return #field in v; // expect `return in v` so runtime will have a syntax error
+}
+
+export { }
+
+
+//// [privateNameInInExpressionTransform.js]
+var __classPrivateFieldIn = (this && this.__classPrivateFieldIn) || function(receiver, state) {
+    if (receiver === null || (typeof receiver !== "object" && typeof receiver !== "function")) throw new TypeError("Cannot use 'in' operator on non-object");
+    return typeof state === "function" ? receiver === state : state.has(receiver);
+};
+var _Foo_instances, _a, _Foo_field, _Foo_method, _Foo_staticField, _Foo_staticMethod, _Bar_field;
+class Foo {
+    constructor() {
+        _Foo_instances.add(this);
+        _Foo_field.set(this, 1);
+    }
+    check(v) {
+        __classPrivateFieldIn(v, _Foo_field); // expect Foo's 'field' WeakMap
+        __classPrivateFieldIn(v, _Foo_instances); // expect Foo's 'method' WeakSet
+        __classPrivateFieldIn(v, _a); // expect Foo's constructor
+        __classPrivateFieldIn(v, _a); // expect Foo's constructor
+    }
+    precedence(v) {
+        // '==' has lower precedence than 'in'
+        // '<'  has same precedence than 'in'
+        // '<<' has higher precedence than 'in'
+        v == __classPrivateFieldIn(v, _Foo_field) == v; // Good precedence: ((v == (#field in v)) == v)
+        v << __classPrivateFieldIn(v << v, _Foo_field); // Good precedence: (v << (#field in (v << v)))
+        v << __classPrivateFieldIn(v, _Foo_field) == v; // Good precedence: ((v << (#field in v)) == v)
+        v == __classPrivateFieldIn(v, _Foo_field) < v; // Good precedence: (v == ((#field in v) < v))
+        __classPrivateFieldIn(v, _Foo_field) && __classPrivateFieldIn(v, _Foo_field); // Good precedence: ((#field in v) && (#field in v))
+    }
+    invalidLHS(v) {
+        'prop' in v;
+        10;
+        __classPrivateFieldIn(v, _Foo_field);
+        10;
+    }
+}
+_a = Foo, _Foo_field = new WeakMap(), _Foo_instances = new WeakSet(), _Foo_method = function _Foo_method() { }, _Foo_staticMethod = function _Foo_staticMethod() { };
+_Foo_staticField = { value: 2 };
+class Bar {
+    constructor() {
+        _Bar_field.set(this, 1);
+    }
+    check(v) {
+        __classPrivateFieldIn(v, _Bar_field); // expect Bar's 'field' WeakMap
+    }
+}
+_Bar_field = new WeakMap();
+function syntaxError(v) {
+    return  in v; // expect `return in v` so runtime will have a syntax error

```

Comment:
You can leave `return #priv in v`. It will also be a runtime syntax error.

Reply:
the current implementation matches the existing behavior for private field access where it would be a syntax error.

Reply:
To be clearer, even though this PR has a test for this. The reason this happens is because this PR doesn't have a special case for this, and 'falls through' to the existing emit behavior. Ref: https://github.com/bloomberg/TypeScript/blob/61c677be3e7fa89d18f26158f5b1825c83feb747/src/compiler/transformers/classFields.ts#L297-L298